api/backend/blueflood/blueflood.go

```diff
@@ -132,20 +140,61 @@ func (b *blueflood) FetchSingleSeries(request api.FetchSeriesRequest) (api.Times
 	}
 	queryResolution := b.config.bluefloodResolution(
 		request.Timerange.Resolution(),
-		request.Timerange.Start())
+		request.Timerange.Start(),
+	) // The query resolution will be downsampled automatically based on the timerange?
 
+	// <Sample the data>
 	queryUrl, err := b.constructURL(request, sampler, queryResolution)
 	if err != nil {
 		return api.Timeseries{}, err
 	}
-
-	// Issue GET to fetch metrics
 	parsedResult, err := b.fetch(request, queryUrl)
 	if err != nil {
 		return api.Timeseries{}, err
 	}
+	// </Sample the data>
```

Comment:
Probably don't need a comment here about the end of a logical block. You can add a comment at the beginning and it should be fine.